python/astro_metadata_translator/tests.py

```diff
@@ -149,8 +151,29 @@ def assertObservationInfoFromYaml(self, file, dir=None, check_wcs=True,  # noqa:
             inconsistent.
         """
         header = read_test_file(file, dir=dir)
-        self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
-                                   wcs_params=wcs_params, **kwargs)
+
+        try:
+            self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
+                                       wcs_params=wcs_params, **kwargs)
+        except AssertionError as error:
+            raise AssertionError("ObservationInfo derived from an dict-like "
+                                 "type is inconsistent.") from error
+
+        # Explicitly verify consistency when instantiating from Header
+        if not isinstance(header, dict):
+            header = header.toDict()
+        # Header can not parse multi-line header keys as lists
+        for key, val in header.items():
+            if isinstance(val, list):
+                header[key] = "".join(val)
+
+        header = Header(header)
+        try:
+            self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
+                                       wcs_params=wcs_params, **kwargs)
+        except AssertionError as error:
```

Comment:
In LSST code we usually say `as e` for the error catching so please switch to that.